angular-dfp.js

```diff
@@ -429,10 +427,18 @@ angular.module('ngDfp', [])
                 return;
               }
 
-              $timeout(function () {
+              timeoutPromise = $timeout(function () {
                 DoubleClick.refreshAds(id);
               }, scope.timeout);
             });
+
+            // Cancel $interval and $timeout service when DOM destroy
+            element.on('$destroy', function() {
```

Comment:
Actually, scope.$on is working too.

So, According to reference, use element.on at directive, use $scope.$on at controller meybe.
https://docs.angularjs.org/api/ng/service/$interval